install.sh

```diff
@@ -1,243 +1,243 @@
+    else
+        echo "Error: Installation failed."
+        cleanup
+        exit 1
+    fi
+}
+
+# Function to prompt user for confirmation
+user_input() {
+    echo "$1"
+    while true; do
+        printf "Do you want to %s? (Y/n): " "$2"
+        read -r confirm
+        case "$confirm" in
+        [Yy]* | '' | ' ') break ;;
+        [Nn]*) exit 0 ;;
+        *) echo "Invalid input. Please enter 'y' or 'n'." ;;
+        esac
+    done
+}
+
+# Function to uninstall Clipper
+uninstall() {
+    if command -v clipper >/dev/null 2>&1; then
+        echo "Uninstalling $(clipper -v)..."
+        if ! sudo rm -vf "$(command -v clipper)"; then
+            echo "Failed to uninstall."
+            exit 1
+        fi
+        cleanup
+        echo "Done."
+        exit 0
+    else
+        echo "Clipper is not installed. Exiting..."
+        exit 0
+    fi
+}
+
+# Function to log usage message
+log_usage_message() {
+    echo "Usage:"
+    echo " $1 [options]"
+    echo "Options:"
+    echo "    -y  attempts to continue the installation without prompts."
+    echo "    -h  display this usage message."
+    echo "    -r  uninstall Clipper."
+    echo "    -v  display current and latest version of Clipper."
+}
+
+# Function to clean up temporary files
+cleanup() {
+    rm -rf "$TMP_DIR" || {
+        echo "Failed to cleanup temporary files: $?"
+        exit 1
+    }
+}
+
+log_version_message() {
+    get_latest_version
+    get_installed_version
+
+    if [ -n "$INSTALLED_VERSION" ] && [ -n "$LATEST_VERSION" ]; then
+        echo "$INSTALLED_VERSION is currently installed. (Latest: $LATEST_VERSION)"
+        exit 0
+    fi
+
+    if [ -n "$INSTALLED_VERSION" ]; then
+        echo "$INSTALLED_VERSION is currently installed."
+        exit 0
+    fi
+
+    if [ -n "$LATEST_VERSION" ]; then
+        echo "Clipper $LATEST_VERSION is the latest version. To install, run this script: sh $0 -y"
+        exit 0
+    fi
+
+    echo "Failed to get the version of Clipper. See https://github.com/supitsdu/clipper/releases"
+    exit 1
+}
+
+# Main script execution
+while getopts "yhrv" opt; do
+    case $opt in
+    y) AUTO_CONFIRM=true ;;
+    r) uninstall ;;
+    h) log_usage_message "$0" && exit 0 ;;
+    v) log_version_message ;;
+    *) log_usage_message "$0" && exit 1 ;;
+    esac
+done
+
+[ -n "$AUTO_CONFIRM" ] || user_input "This script will install Clipper on your system." "proceed"
+
+get_latest_version
+get_installed_version
+
+if [ -n "$INSTALLED_VERSION" ]; then
+    echo "$INSTALLED_VERSION is already installed."
+    if [ -z "$AUTO_CONFIRM" ]; then
+        if is_newer_version "$LATEST_VERSION" "$INSTALLED_VERSION"; then
+            user_input "A newer version ($LATEST_VERSION) is available." "upgrade"
+        else
+            user_input "You have the latest version installed." "reinstall"
+        fi
+    fi
+fi
+
+set_binary_url
+download_binary
+make_executable
+install_binary
+verify_installation
+
+# Clean up temporary directory
+cleanup

```

Comment:
So no trap yet?

Reply:
It was an error see https://github.com/supitsdu/clipper/pull/19#issuecomment-2188198610